src/main/scala/dotty/dokka/tasty/comments/Queries.scala

```diff
@@ -0,0 +1,159 @@
+package dotty.dokka.tasty.comments
+
+sealed trait Query {
+  def asList: List[String] = this match {
+    case Query.StrictMemberId(id) => id :: Nil
+    case Query.Id(id) => id :: Nil
+    case Query.QualifiedId(qual, _, rest) => qual.asString :: rest.asList
+  }
+
+  def join: String =
+    def go(sb: StringBuilder, segment: Query): String = segment match {
+      case Query.StrictMemberId(id) =>
+        sb ++= id
+        sb.toString
+      case Query.Id(id) =>
+        sb ++= id
+        sb.toString
+      case Query.QualifiedId(qual, sep, rest) =>
+        sb ++= qual.asString
+        sb += sep
+        go(sb, rest)
+    }
+    go(new StringBuilder, this)
+}
+
+sealed trait QuerySegment extends Query
+object Query {
```

Comment:
I think a few lines of doc would be worth it here, so that our future selves know immediately which `[[]]` comment corresponds to which type of query.

Reply:
This is still subject to change, I'm continuing to mull over exactly how we want to support linking to documentation. Scaladoc syntax manages to simultaneously be ugly (overloading resolution looks horrible), cryptic (_terminating_ names in `!` or `$` distinguishes between types and terms, and terminating in `*` is also possible and _may_ actually do something) and redundant (qualifier can be separated from rest of ID by either `.` or `#`, and those mean exactly the same). I'd like to change that, but first I think I'd like to see how existing projects use links, so that I have an idea of what to optimise for.

Also, we have fairly succinct tests for parsing, so that's one way one can tell what syntax corresponds to what representation.

Reply:
okay, fair enough